bench-vote/src/main.rs

```diff
@@ -0,0 +1,331 @@
+                .help("Use this many producer threads."),
+        )
+        .arg(
+            Arg::with_name("server-address")
+                .short('n')
+                .long("server-address")
+                .value_name("HOST:PORT")
+                .takes_value(true)
+                .validator(|arg| solana_net_utils::is_host_port(arg.to_string()))
+                .help("The destination streamer address to which the client will send transactions to"),
+        )
+        .arg(
+            Arg::new("use-connection-cache")
+                .long("use-connection-cache")
+                .takes_value(false)
+                .help("Use this many producer threads."),
+        )
+        .arg(
+            Arg::new("verbose")
+                .long("verbose")
+                .takes_value(false)
+                .help("Show verbose messages."),
+        )
+        .get_matches();
+
+    if let Some(n) = matches.value_of("num-recv-sockets") {
+        num_sockets = max(num_sockets, n.to_string().parse().expect("integer"));
+    }
+
+    let num_producers: u64 = matches.value_of_t("num-producers").unwrap_or(4);
+
+    let use_connection_cache = matches.is_present("use-connection-cache");
+
+    let server_only = matches.is_present("server-only");
+    let client_only = matches.is_present("client-only");
+    let verbose = matches.is_present("verbose");
+
+    let destination = matches.is_present("server-address").then(|| {
+        let addr = matches
+            .value_of("server-address")
+            .expect("Destination must be set when --client-only is used");
+        solana_net_utils::parse_host_port(addr).expect("Expecting a valid server address")
+    });
+
+    let port = destination.map_or(0, |addr| addr.port());
+    let ip_addr = destination.map_or(IpAddr::V4(Ipv4Addr::UNSPECIFIED), |addr| addr.ip());
+
+    let (exit, read_threads, sin_threads, destination) = if !client_only {
+        let exit = Arc::new(AtomicBool::new(false));
+
+        let mut read_channels = Vec::new();
+        let mut read_threads = Vec::new();
+        let recycler = PacketBatchRecycler::default();
+        let (port, read_sockets) = solana_net_utils::multi_bind_in_range(
+            ip_addr,
+            (port, port + num_sockets as u16),
+            num_sockets,
+        )
+        .unwrap();
+        let stats = Arc::new(StreamerReceiveStats::new("bench-streamer-test"));
+        for read in read_sockets {
+            read.set_read_timeout(Some(Duration::new(1, 0))).unwrap();
+
+            let (s_reader, r_reader) = unbounded();
+            read_channels.push(r_reader);
+            read_threads.push(receiver(
+                "solRcvrBenStrmr".to_string(),
+                Arc::new(read),
+                exit.clone(),
+                s_reader,
+                recycler.clone(),
+                stats.clone(),
+                Duration::from_millis(1), // coalesce
+                true,
+                None,
+                false,
```

Comment:
for the sake of the reader of this code, these need some trailing comments what each are.
```rust
true,  // use_pinned_memory
None,  // in_vote_only_mode
false, //is_staked_service
```

Reply:
done